yjit/src/codegen.rs

```diff
@@ -2536,6 +2541,11 @@ fn gen_setlocal_generic(
     }
 
     let (flags_opnd, local_opnd) = if level == 0 && jit.assume_no_ep_escape(asm) {
```

Comment:
This seems like it would make 2 blocks with the ep escape assumption no? Before and after the defer_compilation(). I think only the newest block should assume the assumption.